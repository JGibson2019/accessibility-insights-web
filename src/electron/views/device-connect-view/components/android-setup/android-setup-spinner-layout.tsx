// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { AndroidSetupActionCreator } from 'electron/flux/action-creator/android-setup-action-creator';
import {
    DeviceConnectFooter,
    DeviceConnectFooterDeps,
} from 'electron/views/device-connect-view/components/device-connect-footer';
import { Spinner, SpinnerSize } from 'office-ui-fabric-react';
import * as React from 'react';

import { androidSetupSpinner } from './android-setup.scss';

export type AndroidSetupSpinnerDeps = {
    androidSetupActionCreator: AndroidSetupActionCreator;
} & DeviceConnectFooterDeps;
export type AndroidSetupSpinnerProps = {
    deps: AndroidSetupSpinnerDeps;
    spinnerLabel: string;
};

export const AndroidSetupSpinnerLayout = NamedFC<AndroidSetupSpinnerProps>(
    'AndroidSetupSpinnerLayout',
    (props: AndroidSetupSpinnerProps) => {
        return (
            <>
                <Spinner
                    className={androidSetupSpinner}
                    size={SpinnerSize.large}
                    label={props.spinnerLabel}
                    role="alert"
                    aria-live={props['aria-live']}
                />
                <DeviceConnectFooter
                    deps={props.deps}
                    cancelClick={props.deps.androidSetupActionCreator.cancel}
                    canStartTesting={false}
                ></DeviceConnectFooter>
            </>
        );
    },
);
