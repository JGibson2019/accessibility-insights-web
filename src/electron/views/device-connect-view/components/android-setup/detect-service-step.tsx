// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
import { AndroidSetupSpinner } from './android-setup-spinner';
import { CommonAndroidSetupStepProps } from './android-setup-types';

export const DetectServiceStep = NamedFC<CommonAndroidSetupStepProps>(
    'DetectServiceStep',
    (props: CommonAndroidSetupStepProps) => {
        return (
            <main>
                <AndroidSetupSpinner label="Scanning for the service on your device..." />
            </main>
        );
    },
);
