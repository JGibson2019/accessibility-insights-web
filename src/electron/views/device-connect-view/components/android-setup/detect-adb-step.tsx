// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { CommonAndroidSetupStepProps } from 'electron/views/device-connect-view/components/android-setup/android-setup-types';
import { Spinner, SpinnerSize } from 'office-ui-fabric-react';
import * as React from 'react';

import { androidSetupSpinner } from '../../device-connect-view.scss';

export const DetectAdbStep = NamedFC<CommonAndroidSetupStepProps>(
    'DetectAdbStep',
    (props: CommonAndroidSetupStepProps) => {
        const renderSpinner = () => {
            return (
                <Spinner
                    className={androidSetupSpinner}
                    size={SpinnerSize.large}
                    label="Loading..."
                    role="alert"
                    aria-live={props['aria-live']}
                />
            );
        };
        return renderSpinner();
    },
);
