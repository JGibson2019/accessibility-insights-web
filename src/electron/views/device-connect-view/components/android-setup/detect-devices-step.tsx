// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { AndroidSetupSpinnerLayout } from 'electron/views/device-connect-view/components/android-setup/android-setup-spinner-layout';
import { CommonAndroidSetupStepProps } from 'electron/views/device-connect-view/components/android-setup/android-setup-types';
import * as React from 'react';

// export type DetectDevicesStepDeps = BufferingDeps;
// export type DetectDevicesStepProps = BufferingProps;

export const DetectDevicesStep = NamedFC<CommonAndroidSetupStepProps>(
    'DetectAdbStep',
    (props: CommonAndroidSetupStepProps) => {
        return (
            <AndroidSetupSpinnerLayout
                deps={props.deps}
                spinnerLabel="Scanning for devices..."
            ></AndroidSetupSpinnerLayout>
        );
    },
);
