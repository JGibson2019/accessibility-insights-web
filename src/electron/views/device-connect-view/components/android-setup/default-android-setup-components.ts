// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AndroidSetupStepComponentProvider } from 'electron/views/device-connect-view/components/android-setup/android-setup-types';
import { DetectAdbStep } from 'electron/views/device-connect-view/components/android-setup/detect-adb-step';
import { DetectDevicesStep } from 'electron/views/device-connect-view/components/android-setup/detect-devices-step';

export const defaultAndroidSetupComponents: AndroidSetupStepComponentProvider = {
    'detect-adb': DetectAdbStep,
    'detect-devices': DetectDevicesStep,
};
