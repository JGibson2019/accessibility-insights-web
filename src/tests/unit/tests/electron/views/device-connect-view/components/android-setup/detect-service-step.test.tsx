// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { CommonAndroidSetupStepProps } from 'electron/views/device-connect-view/components/android-setup/android-setup-types';
import { DetectServiceStep } from 'electron/views/device-connect-view/components/android-setup/detect-service-step';
import { shallow } from 'enzyme';
import * as React from 'react';
describe('DetectServiceStep', () => {
    const props: CommonAndroidSetupStepProps = {
        userConfigurationStoreData: null,
        androidSetupStoreData: {
            currentStepId: 'detect-service',
        },
        deps: {
            androidSetupActionCreator: null,
            androidSetupStepComponentProvider: null,
            LinkComponent: null,
        },
    };

    it('renders', () => {
        const rendered = shallow(<DetectServiceStep {...props} />);
        expect(rendered.getElement()).toMatchSnapshot();
    });
});
