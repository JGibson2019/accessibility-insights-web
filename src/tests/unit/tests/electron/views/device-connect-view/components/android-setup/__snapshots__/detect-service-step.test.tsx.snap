// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetectServiceStep renders 1`] = `
<main>
  <AndroidSetupSpinnerStep
    deps={
      Object {
        "LinkComponent": null,
        "androidSetupActionCreator": null,
        "androidSetupStepComponentProvider": null,
      }
    }
    disableCancel={true}
    spinnerLabel="Scanning for devices..."
  />
</main>
`;
